### DISCUSSION TITLE
MIC plot theme

### DISCUSSION THREAD

**Discussion by theanita1** [msberends/AMR#98] [Q&A] [answered]
Hi Matthijs (or anyone else who can answer my question)

On the website the MIC plots display quite nicely - white background, black around the bars etc, 
However when I use the autoplot function it comes back with the typical R formatting

Should I need additional ggplot coding to achieve the same output as what's displayed on the instructions? 
_edit_
If yes would you mind sharing the additional required code?
I can't manage to get the same output with my current ggplot knowledge

**Comment by msberends** [comment]
Dear @theanita1,

There is no additional code required, since the website uses exactly the code that precede the plots (that should at least be the case...). So, for example:

```r
my_mics <- random_mic(100)
ggplot2::autoplot(my_mics,
                  mo = "Escherichia coli",
                  ab = "ampicillin",
                  guideline = "EUCAST 2021")
```

<img width="802" alt="image" src="https://user-images.githubusercontent.com/31037261/224543725-b33f63ff-c870-4db4-8b27-0780520e69c6.png">

Can you reproduce this? Which plots are you referring to from the website?

**Reply by theanita1** [reply]
Hi Matthijs,
I'm referring to the plots on [this page](https://msberends.github.io/AMR/reference/plot.html)
Particularly the ones that look like this:
![image](https://user-images.githubusercontent.com/11519951/224614828-be6f0865-d6c8-457d-9bc8-1d442a4c16d1.png)
The x-axis is removed, the y-axis is clean, the background is plain and the boxes are outlined. Additionally the legend spans across the top rather than on the side of the graph

**Reply by msberends** [reply, marked as answer]
Ahhh, but that's not ggplot 😉 That's base R.

I wrote something to get to similar results in ggplot for you:

```r
ggplot2::autoplot(random_mic(100),
                  mo = "Escherichia coli",
                  ab = "ampicillin",
                  guideline = "EUCAST 2021") +
  # here starts the theme part
  ggplot2::theme_minimal() + 
  ggplot2::theme(panel.grid = ggplot2::element_blank(), 
                 axis.line.y = ggplot2::element_line(linewidth = 0.25), 
                 legend.position = "top")
```

<img width="778" alt="image" src="https://user-images.githubusercontent.com/31037261/224751218-9db1c307-7123-411b-aa30-12068eaf91f3.png">


Or you could create your own function to reduce code:

```r
our_theme <- function(...) {
  ggplot2::theme_minimal() + 
    ggplot2::theme(panel.grid = ggplot2::element_blank(), 
                   axis.line.y = ggplot2::element_line(linewidth = 0.25), 
                   legend.position = "top",
                   ...)
}

# and now just add our_theme() to your future code (works with any ggplot):
ggplot2::autoplot(random_mic(100),
                  mo = "Escherichia coli",
                  ab = "ampicillin",
                  guideline = "EUCAST 2021") +
  our_theme()
```

**Reply by theanita1** [reply]
Thanks for that - still learning the in's and out's of R so I appreciate the feedback and code!
